pkg/agent/controller/egress/egress_controller.go

```diff
@@ -223,6 +239,15 @@ func (c *EgressController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	if c.shouldStartCluster() {
```

Comment:
This won't work if a pool is created after the code is run. I think the cluster should be always started, but the node itself will join or leave the cluster based on ExternalIPPool and Node events.

Reply:
done